activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -1108,6 +1108,17 @@ def arel_columns(columns)
     end
 
     def reverse_sql_order(order_query)
+      if offset_index != 0
+        ActiveSupport::Deprecation.warn(<<-WARNING.squish)
+            Finding a last element by loading the relation when SQL ORDER
```

Comment:
Oh, you're right that the message doesn't quite fit. I suppose that because in this case it is being triggered in the `reverse_sql` query method, the right message is `Reversing the SQL ORDER of a relation with offset applied is deprecated.` I've made that change in 5e75331.